packages/flutter_tools/lib/src/device.dart

```diff
@@ -249,10 +250,59 @@ class DeviceManager {
             .where((Device device) => device.ephemeral == true)
             .toList();
       }
+      // If it was not able to prioritize a device. For example, if the user
+      // has two active Android devices running, then we request the user to
+      // choose one.
+      if (devices.length > 1 && _allEphemeral(devices)){
+        globals.printStatus(globals.userMessages.flutterMultipleDevicesFound);
+        await Device.printDevices(devices);
+        final Device chosenDevice = await _chooseOneOfAvailableDevices(devices);
+        deviceManager.specifiedDeviceId = chosenDevice.id;
+        devices = <Device>[chosenDevice];
+      }
     }
     return devices;
   }
 
+  bool _allEphemeral(List<Device> devices) {
+      return devices.every((Device device) => device.ephemeral == true);
+  }
+
+  Future<Device> _chooseOneOfAvailableDevices(List<Device> devices) async {
+    _displayDeviceOptions(devices);
+    final String userInput =  await _readUserInput(devices.length);
+    return devices[int.parse(userInput)];
```

Comment:
this will throw a formatException if the result is not a valid number. Use `int.tryParse` and check for null. On an invalid result, either exit or prompt again

Reply:
I'm a little confused here, as I understand, promptForCharInput from the terminal class already takes care of this with the acceptedCharacters list. When an invalid input is provided by the user this already prompt again. So I don't see how this could generate an invalid result. Am I mistaken?

Reply:
Ahh you're right, good catch